### PR TITLE
Add documentation of the PostgreSQL Update Strategy

### PR DESCRIPTION
**Pull request by glrf**
Adds the documentation of the UpdateStrategy feature implemented in https://github.com/vshn/component-appcat/pull/137 and https://github.com/vshn/appcat-comp-functions/pull/23

### CODE REVIEW

File: docs/modules/ROOT/pages/vshn-managed/postgresql/update-strategy.adoc [hunk 1]
```diff
@@ -0,0 +1,48 @@
+= Instance Update Strategy
+
+Certain changes to the PostgreSQL instance spec requires a restart of the database to take effect.
+Most importantly changes to the `size` of the instance will need a restart of the instance to apply.§
```
Comment (Kidswiss) [inline docs/modules/ROOT/pages/vshn-managed/postgresql/update-strategy.adoc:4]:
```suggestion
Most importantly changes to the `size` of the instance will need a restart of the instance to apply.
```

File: docs/modules/ROOT/pages/vshn-managed/postgresql/update-strategy.adoc [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+An instance's `.spec.parameters.updateStrategy` field allows you to configure and disable automated retarts after an upated of the PostgreSQL instance.
+There are two possible values:
+
+.Immediate
+
+The `Immediate` strategy will automatically restart the database if one is necessary for the updates to the instance to apply.
+This is the default update strategy.
+
+[IMPORTANT]
+Please be aware for single instance databases, this might cause a short downtime after updating.
```
Comment (Kidswiss) [inline docs/modules/ROOT/pages/vshn-managed/postgresql/update-strategy.adoc:15]: I'd argue that's also true for instance with multiple pods, as there will be some sort of failover via patroni. But I'd expect that to be much shorter.

**Review by Kidswiss — CHANGES_REQUESTED** [review]
One typo and some input.

Also just had a thought: should we also expose the flag to control the restart method in the future? (`InPlace` and `ReducedImpact`)

File: docs/modules/ROOT/pages/vshn-managed/postgresql/update-strategy.adoc (hunk 2 above)
Comment (glrf) [reply docs/modules/ROOT/pages/vshn-managed/postgresql/update-strategy.adoc:15]: Fair. But I expect that to basically only be a connection reset. So I don't know if I'd actually call that "downtime". But you're right I'll rephrase.

**Comment by glrf** [comment]
> Also just had a thought: should we also expose the flag to control the restart method in the future? (`InPlace` and `ReducedImpact`)

I somehow had issues getting the `ReducedImpact` to work properly so I did not implement it in the first iteration. Could be something to look into if there is demand but did not want to invest too much time into it right now.

**Review by Kidswiss — APPROVED** [review]

File: docs/modules/ROOT/pages/references/crds.adoc [hunk 3]
```diff
@@ -797,6 +778,8 @@ VSHNPostgreSQL is the API for creating Postgresql clusters.
 | Field | Description
 | *`schedule`* __string__ | 
 | *`retention`* __integer__ | 
+| *`deletionProtection`* __boolean__ | DeletionProtection will protect the instance from being deleted for the given retention time. This is enabled by default.
+| *`deletionRetention`* __integer__ | DeletionRetention specifies in days how long the instance should be kept after deletion. The default is keeping it one week.
```
Comment (zugao) [inline docs/modules/ROOT/pages/references/crds.adoc:782]:
```suggestion
| *`deletionRetention`* __integer__ | DeletionRetention specifies in days how long the instance should be kept after deletion. The default value is one week.
```

File: docs/modules/ROOT/pages/references/crds.adoc [hunk 4]
```diff
@@ -876,6 +859,7 @@ VSHNPostgreSQLParameters are the configurable fields of a VSHNPostgreSQL.
 | *`restore`* __xref:{anchor_prefix}-github-com-vshn-component-appcat-apis-vshn-v1-vshnpostgresqlrestore[$$VSHNPostgreSQLRestore$$]__ | Restore contains settings to control the restore of an instance.
 | *`monitoring`* __xref:{anchor_prefix}-github-com-vshn-component-appcat-apis-vshn-v1-vshnpostgresqlmonitoring[$$VSHNPostgreSQLMonitoring$$]__ | Monitoring contains settings to control monitoring.
 | *`encryption`* __xref:{anchor_prefix}-github-com-vshn-component-appcat-apis-vshn-v1-vshnpostgresqlencryption[$$VSHNPostgreSQLEncryption$$]__ | Encryption contains settings to control the storage encryption of an instance.
+| *`updateStrategy`* __xref:{anchor_prefix}-github-com-vshn-component-appcat-apis-vshn-v1-vshnpostgresqlupdatestrategy[$$VSHNPostgreSQLUpdateStrategy$$]__ | UpdateStrategy indicates how and when updates to the instance spec will be applied.
```
Comment (zugao) [inline docs/modules/ROOT/pages/references/crds.adoc:862]: If I understood correctly it indicates just "when".

File: docs/modules/ROOT/pages/references/crds.adoc [hunk 5]
```diff
@@ -923,19 +907,38 @@ VSHNPostgreSQLSpec defines the desired state of a VSHNPostgreSQL.
...
+
+.Appears In:
+****
+- xref:{anchor_prefix}-github-com-vshn-component-appcat-apis-vshn-v1-vshnpostgresqlparameters[$$VSHNPostgreSQLParameters$$]
+****
+
+[cols="25a,75a", options="header"]
+|===
+| Field | Description
+| *`type`* __string__ | Type indicates the type of the UpdateStrategy. Default is OnDelete. Possible enum values: - `"OnRestart"` indicates that the changes to the spec will only be applied once the instance is restarted by other means, most likely during maintenance. - `"Immediate"` indicates that update will be applied to the instance as soon as the spec changes. Please be aware that this might lead to short downtime.
```
Comment (zugao) [inline docs/modules/ROOT/pages/references/crds.adoc:938]: I thought we agreed only 2 types of `UpdateStrategy`.

File: docs/modules/ROOT/pages/vshn-managed/postgresql/update-strategy.adoc [hunk 6]
```diff
@@ -0,0 +1,48 @@
+= Instance Update Strategy
+
+Certain changes to the PostgreSQL instance spec requires a restart of the database to take effect.
+Most importantly changes to the `size` of the instance will need a restart of the instance to apply.
+
+An instance's `.spec.parameters.updateStrategy` field allows you to configure and disable automated retarts after an upated of the PostgreSQL instance.
```
Comment (zugao) [inline docs/modules/ROOT/pages/vshn-managed/postgresql/update-strategy.adoc:6]:
```suggestion
An instance's `.spec.parameters.updateStrategy` field allows you to configure and disable automated restarts after an update of the PostgreSQL instance.
```

File: docs/modules/ROOT/pages/vshn-managed/postgresql/update-strategy.adoc [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+There are two possible values:
+
+.Immediate
+
+The `Immediate` strategy will automatically restart the database if one is necessary for the updates to the instance to apply.
```
Comment (zugao) [inline docs/modules/ROOT/pages/vshn-managed/postgresql/update-strategy.adoc:11]:
```suggestion
The `Immediate` strategy will automatically restart the database whenever the instance is changed.
```

File: docs/modules/ROOT/pages/vshn-managed/postgresql/update-strategy.adoc [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    size:
+      cpu: "600m"
+      memory: "3500Mi"
+      disk: "80Gi"
+    updateStrategy:
+      type: OnRestart <1>
+  writeConnectionSecretToRef:
+    name: postgres-creds
+----
+<1> The update strategy `OnRestart` will never trigger a restart after an update
```
Comment (zugao) [inline docs/modules/ROOT/pages/vshn-managed/postgresql/update-strategy.adoc:48]:
```suggestion
<1> The update strategy `OnRestart` will never trigger an immediate restart after update.
```

**Review by zugao — COMMENTED** [review]
Some things to change. Overall good.

**Review by zugao — COMMENTED** [review]
Some things to change. Overall good.

File: docs/modules/ROOT/pages/references/crds.adoc (hunk 3 above)
Comment (glrf) [reply docs/modules/ROOT/pages/references/crds.adoc:782]: This is a completely unrelated change and is auto generated. I won't change this in this PR

**Pull request merged**